### ISSUE TITLE
SetWebApp wipes AppSettings after suppressed exception from GetWebAppConfiguration

### ISSUE THREAD

**Issue by Ralms** [Azure/azure-powershell#19942]
### Description

This is a silent bug due to the fact that GetWebApp section, where it retrieves the AppSettings, is absorbing all exceptions. 

While using a service account which only has Global Read + PIM, there is the possibility where the App Settings for an app are wiped if the PIM permission request hasn't fully replicated yet. 

Timeline: 

1. Elevate permissions with a PIM request to have Contribute temporarily.
2. Perform Set-AZWebApp -ResourceGroupName MyRGName -Name MyWebAppName -FtpsState FtpsOnly
3. Check WebApp which has no AppSettings as they were cleared. 

The [SetAzureWebApp Cmdlet](https://github.com/Azure/azure-powershell/blob/main/src/Websites/Websites/Cmdlets/WebApps/SetAzureWebApp.cs#L154) gets the App configuration first using WebsitesClient [GetWebApp](https://github.com/Azure/azure-powershell/blob/76c8995ba80e5976d421003be1922490e0de1261/src/Websites/Websites/Utilities/WebsitesClient.cs#L247), which retrieves the [AppSettings](https://github.com/Azure/azure-powershell/blob/76c8995ba80e5976d421003be1922490e0de1261/src/Websites/Websites/Utilities/WebsitesClient.cs#L611) on the GetWebAppConfigurationMethod. 

Will than use whatever it got to do some changes (in case of container changes) and resubmit the AppSettings. 

The Issue resides in the fact that GetWebAppConfiguration is blindly absorbing all exceptions, removing all control from higher level methods. 
This exception should be thrown and handled on the cmdlets themselves. 

Please note that this was discovered on a bulk operation over dozens of apps, where FTP mode was changed and just around 10 apps lost their AppSettings. 

### Issue script & Debug output

```PowerShell
# This bug only manifests itself within specific timings between PIM permissions replicating and executing the several commands. 
Set-AZWebApp -ResourceGroupName MyRGName -Name MyWebAppName -FtpsState FtpsOnly
```

### Environment data

```PowerShell
NA
```

### Module versions

```PowerShell
NA
```

### Error output

```PowerShell
Silent error due to code absorbing the exception.
```

**Comment by Kotasudhakarreddy** [comment]
@Ralms Can you please update the issue with **Module Versions** and **Environment Data**

**Comment by Ralms** [comment]
@Kotasudhakarreddy Could you please elaborate on the relevance in the module Versions and Environment Data considering I explained where the bug in the code is located? Doesn't really matter where the cmdlet is called, as long as it's using the last available version. 

Is a logic issue, where exceptions are being silently absorbed, causing bad behavior higher up in the code. 

I've already shared links with references to the lines, showing the code flow to the where the issue happens. 
Still, allow me to share it once more. 

[SetAzureWebApp Cmdlet calls WebsitesClient.GetWebApp](https://github.com/Azure/azure-powershell/blob/main/src/Websites/Websites/Cmdlets/WebApps/SetAzureWebApp.cs#L154) 

[WebsitesClient.GetWebApp() calls WebsitesClient.GetWebAppConfiguration()](https://github.com/Azure/azure-powershell/blob/76c8995ba80e5976d421003be1922490e0de1261/src/Websites/Websites/Utilities/WebsitesClient.cs#L256)

[WebsitesClient.GetWebAppConfiguration() contains a try catch absorbing exceptions blindly](https://github.com/Azure/azure-powershell/blob/76c8995ba80e5976d421003be1922490e0de1261/src/Websites/Websites/Utilities/WebsitesClient.cs#L632)

I can't think of any good scenario where suppressing exceptions like this is good and doesn't cause bugs / bad behaviors. 
And in this case, was yet another example why it should not be done, affecting several apps of a Microsoft customer. 

Let me know if you have any other question.

**Comment by Ralms** [comment]
Fixed on PR #20159
Closing.
